x/genaccounts/internal/types/genesis_account.go

```diff
@@ -15,6 +17,7 @@ import (
 // GenesisAccount is a struct for account initialization used exclusively during genesis
 type GenesisAccount struct {
 	Address       sdk.AccAddress `json:"address" yaml:"address"`
+	PubKey        crypto.PubKey  `json:"public_key" yaml:"public_key"`
```

Comment:
this is a breaking change right? If so, changelog should be updated

Reply:
yeah, Idk why we didn't have the `PubKey` on the genaccounts before. The sims break if I don't define the pubkey for each account. cc: @zmanian @rigelrozanski @cwgoes @alexanderbez @jackzampolin

Reply:
in any case I think we want to get rid of the genaccounts module altogether, see #4986.

Reply:
The public key could be revealed when the first transaction was sent, I believe.

Reply:
> The public key could be revealed when the first transaction was sent, I believe.

@cwgoes that was my thought as well. But the ante handler requires the pubkey (in this case of the simulation accounts) for signature verification. The simulations (`sim_import_export` and `sim_after_import`) won't work without defining the pubkeys on `GenesisAccount`.

Reply:
The simulation doesn't validate signatures. Why would it need a public key? Seems like their needs to be some kind of mock pubkey logic in simulation

Reply:
We now use the full baseapp instead of the mocked handlers. This is one of the error diff msgs from `sim_import_export` if I remove the pubkey from the `GenesisAccount`:

```
Account:
    Address:       cosmos1qa5dwz94a5ealsqt8jn4d4m7j2zhdqthlktlen
    Pubkey:        cosmospub1addwnpepqfad97vsrxgysfr76ag89x7vphfahr7lmnjh6y0t9xxv3ev57e9mvcxqsa6
    Coins:         1167036376stake
    AccountNumber: 0
    Sequence:      95
        	            	
Account:
    Address:       cosmos1qa5dwz94a5ealsqt8jn4d4m7j2zhdqthlktlen
    Pubkey:        
    Coins:         1167036376stake
    AccountNumber: 0
    Sequence:      95Account:
    Address:       cosmos1p8jy7qpaqm4jet8mhuys58ur7d59wa6ku532kq
    Pubkey:        cosmospub1addwnpepqwv6rymjlfw9rxeffqqne7v5rkmytz66yk995q0ym09qqqfyajha5nvknx0
    Coins:         1009937stake
    AccountNumber: 8
    Sequence:      85
```

Reply:
@zmanian @cwgoes the pubkey is set on the ante handler (which as I said wasn't being used on the prev simulation implementation). This is why it's required to add the `PubKey` to the `GenesisAccount` for the simulations to pass:

https://github.com/cosmos/cosmos-sdk/blob/c82ff355a53b193f7484c9988059108951e228be/x/auth/ante/ante.go#L213

Reply:
Agree that we should ideally avoid this change. With [ADR 011](https://github.com/cosmos/cosmos-sdk/pull/4986), we'll have simple `GenesisAccount` interfaces instead of a concrete type which would make this change non-contentious.

Reply:
@fedekunze I believe this PR should be blocked until https://github.com/cosmos/cosmos-sdk/pull/5017 is reviewed and merged. Furthermore, we should create a custom genesis type here and thus this PR is not state-machine breaking.

Reply:
> Furthermore, we should create a custom genesis type here and thus this PR is not state-machine breaking.

if we do that then applications won't be able to declare their own account types bc the GenesisAccount will be fixed and customized only to fulfill the SDK account types.

Reply:
mhhh, I think simulation will need to work with `GenesisAccount` interface and not any exact concrete type/implementations? Let's get back to this once ADR 11 is merged.